docs/file-formats/colors.md

```diff
@@ -2,26 +2,123 @@
 
 This file defines the design system's colors.
 
-### Specification
+## Specification
 
-At the top level, the file is an object containing a `"colors"` array.
+At the top level, the file is an object containing:
+
+- a `theme` array
+- a `colors` array
+
+### `theme`
+
+The `theme` array is optional. The default theme is the first item of the array, if any.
+
+Each theme has the following attributes:
+
+| Attribute  | Type     | Required | Description                                                                                                                                                                                                                                                                                                            |
+| ---------- | -------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `id`       | `string` | Yes      | The unique `id` of the theme, which will be used internally by Lona Studio and in the generated code. This should be _code-friendly_: it must not contain spaces or unusual characters, since it will be used as a variable name. It should also be strictly longer than 1 character and cannot be equal to `extends`. |
+| `name`     | `string` | Yes      | The human-readable name of the theme. This will be displayed in the Lona Studio UI. This will not appear in the generated code.                                                                                                                                                                                        |
+| `comment`  | `string` | No       | An optional description of the theme, explaining contextual information, such as how it should be used. This may be displayed in the Lona Studio UI. This will not appear in the generated code.                                                                                                                       |
+| `metadata` | `object` | No       | Additional data used by the generated code to match platform specific features (for example `metadata.isMacOSDarkTheme` indicates that the theme should be used when the macOS system theme is `dark`).                                                                                                                |
```

Comment:
We do want a layer of indirection yes but the default behavior of macOS is to indeed control which theme is used.

I was thinking that the boilerplate code could be auto-generated when `metadata.isMacOSDarkTheme` is set but didn't really think it through so it might not make sense

Reply:
l like the idea, we can think it through a bit more when we get to the code-gen for this stuff